### PR TITLE
Fixed pfcmd ipmachistory

### LINKED ISSUE #1794: pfcmd ipmachistory: Fix and improve syntax
Current pfcmd ipmachistory command is bogus in the sense that the documented syntax is not working.
Also, need to handle the new "limit" parameter

### PR DESCRIPTION
**Pull request by dwlfrth**
# Description

Fixing pfcmd ipmachistory command + documentation
# Impacts

pfcmd
# Issue

fixes #1794 
# Delete branch after merge

YES
# NEWS file entries
## Bug Fixes
- pfcmd ipmachistory: Fix and improve syntax (#1794)

### CODE REVIEW

**Comment by dwlfrth** [comment]
Review me

**Comment by dwlfrth** [comment]
Bump

**Comment by julsemaan** [comment]
The format of the parameters doesn't realy fit pfcmd's style

For consistency, we shouln't be using arguments prefixed by `--` IMO

Once we're ready, we should change all the commands but for now I think we should keep it consistent

**Comment by julsemaan** [comment]
Here is an example of what we talked:

```
my %params = (
  start_date => undef,
  end_date => undef,
);
foreach my $param (@params) {
  my @data = split('=', $param);
  if(exists($params{$data[0]})) {
    if(len($data[1]) > 1) {
      $params{$data[0]} = $data[1];
    }
   else {
     print "Invalid parameter value ".$data[1];
   }
  }
  else {
    print "unknown param ".$data[0];
  }
}
```

**Comment by dwlfrth** [comment]
Review me

File: lib/pf/cmd/pf/ipmachistory.pm [hunk 1]
```diff
@@ -1,45 +1,83 @@
...
+        'limit'         => undef,
+    );
+    foreach my $param ( @params ) {
+        my @data = split('=', $param);
+        if ( exists($params{$data[0]}) ) {
+            if ( length($data[1]) >= 1 ) {
+                $params{$data[0]} = $data[1];
+            }
+            else {
+                print "Invalid parameter value '$data[1]' for parameter '$data[0]'";
```
Comment (julsemaan) [inline lib/pf/cmd/pf/ipmachistory.pm]: You should return an error here
And print to STDERR

File: lib/pf/cmd/pf/ipmachistory.pm [hunk 2]
```diff
@@ -1,45 +1,83 @@
...
+            }
+        }
+        else {
+            print "Unknown parameter '$data[0]'";
```
Comment (julsemaan) [inline lib/pf/cmd/pf/ipmachistory.pm]: You should return an error here
And print to STDERR

File: lib/pf/cmd/pf/ipmachistory.pm [hunk 3]
```diff
@@ -1,45 +1,83 @@
...
-        $params{'end_time'} = str2time( $end_time) if defined $end_time;
-        $self->{params} = \%params;
-        $self->{key} = $key;
+        my $function = \&pf::iplog::get_archive;
+        $params{'start_time'} = str2time( $params{'start_time'}) if defined $params{'start_time'};
+        $params{'end_time'} = str2time( $params{'end_time'}) if defined $params{'end_time'};
         $self->{function} = $function;
+        $self->{key} = $key;
+        $self->{params} = \%params;
         return 1;
```
Comment (julsemaan) [inline lib/pf/cmd/pf/ipmachistory.pm:78]: constant ;)

File: lib/pf/cmd/pf/ipmachistory.pm [hunk 4]
```diff
@@ -1,45 +1,83 @@
...
     }
+
     return 0;
```
Comment (julsemaan) [inline lib/pf/cmd/pf/ipmachistory.pm:81]: constant ;)

**Comment by julsemaan** [comment]
After those we're done and we merge

**Pull request merged**